pkg/apis/security/types_tokenrequest.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package security
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// TokenRequest is a resource that is used to request WorkloadIdentity tokens.
+type TokenRequest struct {
+	metav1.TypeMeta
+	// Standard object metadata.
+	metav1.ObjectMeta
+	// Spec holds configuration settings for the requested token.
+	Spec TokenRequestSpec
+	// Status bears the issued token with additional information back to the client.
+	Status TokenRequestStatus
+}
+
+// TokenRequestSpec holds configuration settings for the requested token.
+type TokenRequestSpec struct {
+	// ContextObject identifies the object the token is requested for.
+	ContextObject *ContextObject
+	// DurationSeconds specifies for how long the requested token should be valid.
+	DurationSeconds int64
```

Comment:
I propose that we align this with the TokenRequest API of Kubernetes.
https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/

```suggestion
	ExpirationSeconds int64
```